### DISCUSSION TITLE
Is it possible to retrieve only new logs after recreating LogWatch after closed connection?

### DISCUSSION THREAD

**Discussion by if-of** [fabric8io/kubernetes-client#5602] [Q&A] [answered]
I use LogWatch to receive real-time logs from pods. However, there is an issue where LogWatch stops receiving logs after 4 hours, and this appears to be linked to a hardcoded 4-hour connection timeout in Kubernetes [(reference here)](https://github.com/kubernetes/kubernetes/blob/f85ac3271c42c7964e86b5200e63e49e525befd4/pkg/kubelet/server/server.go#L168). On the other side, it seems that this timeout can be configured using the streamingConnectionIdleTimeout parameter, as mentioned [(here)](https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration). Unfortunately, AKS does not offer this configuration option.

Is there a way to only retrieve new logs after recreating LogWatch? Currently, when I recreate LogWatch, it restarts log retrieval from the beginning, which can be resource-intensive given the need to re-receive all logs for the past 4 hours.

**Comment by shawkins** [comment, marked as answer]
> Is there a way to only retrieve new logs after recreating LogWatch? Currently, when I recreate LogWatch, it restarts log retrieval from the beginning, which can be resource-intensive given the need to re-receive all logs for the past 4 hours.

There are a couple of parameters supported by the log operation - sinceSeconds, and sinceTime.  I'm not sure there's a better reference for it, so here's from the kube source https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/logs/logs.go#L176

You can set this prior to the watchLog method:

pod.sinceSeconds("4h").watchLog(out)
